sycl/include/CL/sycl/buffer.hpp

```diff
@@ -351,6 +351,54 @@ class buffer {
         impl, reinterpretRange, OffsetInBytes, IsSubBuffer);
   }
 
+  // reinterpret() with no arguments. Can be specialized with one or two
+  // template arguments. four possible combinations two template specializers:
+  // R is same size as T. RD is same as dimensions.
+  // RD is one. To avoid ambiguity, R size is different or previous dimension is
+  // not same.
+
+  // single template specializers
+  // R is same size as T.  RD is assumed to be same as previous.
+  // R is not same as T  previous dimensions were 1.
+
+  template <typename ReinterpretT, int ReinterpretDim>
```

Comment:
Why do you need separate definitions for cases of 1 and 2 template parameters?  The spec just has a default of `ReinterpretDim = dimensions`, so I'm wondering why you can't do the same thing in the implementation.  It seems like that would reduce the amount of code.  Is there something about the SFINAE that doesn't work when the template parameter has a default value?

Reply:
Thanks, that's a lot simpler. I was so focused on making sure the cases were covered I overlooked that.